gulpfile.js

```diff
@@ -148,11 +151,46 @@ function prebidSource(webpackCfg) {
   const analyticsSources = helpers.getAnalyticsSources();
   const moduleSources = helpers.getModulePaths(externalModules);
 
-  return gulp.src([].concat(moduleSources, analyticsSources, 'src/prebid.js'))
+  return gulp.src([].concat(moduleSources, analyticsSources, helpers.getPrecompiledPath('src/prebid.js')))
     .pipe(helpers.nameModules(externalModules))
     .pipe(webpackStream(webpackCfg, webpack));
 }
 
+const PRECOMP_TASKS = new Map();
+function babelPrecomp({distUrlBase = null, disableFeatures = null, dev = false} = {}) {
+  if (dev && distUrlBase == null) {
+    distUrlBase = argv.distUrlBase || '/build/dev/'
+  }
+  const key = `${distUrlBase}::${disableFeatures}`;
+  if (!PRECOMP_TASKS.has(key)) {
+    const babelConfig = require('./babelConfig.js')({
+      disableFeatures: disableFeatures ?? helpers.getDisabledFeatures(),
+      prebidDistUrlBase: distUrlBase ?? argv.distUrlBase
```

Comment:
why and when are those optional?

Reply:
they have a default that can be overridden - only the override is optional.